File: tools/token_bench/programs/05_http_api/http_api.js
const express = require("express");
const app = express();

app.use(express.raw({type: "*/*"}));

app.get("/", (req, res) => res.send("Welcome to API"));
app.get("/health", (req, res) => res.send("ok"));
app.post("/echo", (req, res) => res.send(req.body));
app.use((req, res) => res.status(404).send("not found"));

app.listen(8080);

File: tools/token_bench/programs/05_http_api/http_api.cs
using System.IO;
using System.Net;

class Program {
    static void Main() {
        var listener = new HttpListener();
        listener.Prefixes.Add("http://localhost:8080/");
        listener.Start();

        while (true) {
            var ctx = listener.GetContext();
            var req = ctx.Request;
            var res = ctx.Response;

            string body;
            int status = 200;

            if (req.HttpMethod == "GET" && req.Url.AbsolutePath == "/") {
                body = "Welcome to API";
            } else if (req.HttpMethod == "GET" && req.Url.AbsolutePath == "/health") {
                body = "ok";
            } else if (req.HttpMethod == "POST" && req.Url.AbsolutePath == "/echo") {
                using var reader = new StreamReader(req.InputStream);
                body = reader.ReadToEnd();
            } else {
                body = "not found";
                status = 404;
            }

            res.StatusCode = status;
            using var writer = new StreamWriter(res.OutputStream);
            writer.Write(body);
        }
    }
}
